app/controllers/mixins/actions/vm_actions/right_size.rb

```diff
@@ -6,28 +6,26 @@ def vm_right_size
           assert_privileges(params[:pressed])
           # check to see if coming from show_list or drilled into vms from another CI
           rec_cls = "vm"
-          recs = params[:display] ? find_checked_ids_with_rbac(VmOrTemplate) : [find_id_with_rbac(VmOrTemplate, params[:id]).to_i]
-          if recs.length < 1
-            add_flash(_("One or more %{model} must be selected to Right-Size Recommendations") %
+          record = find_records_with_rbac(VmOrTemplate, checked_or_params).first
+          if record.nil?
+            add_flash(_("One %{model} must be selected to Right-Size Recommendations") %
               {:model => ui_lookup(:table => request.parameters[:controller])}, :error)
             @refresh_div = "flash_msg_div"
             @refresh_partial = "layouts/flash_msg"
             return
-          else
-            if VmOrTemplate.includes_template?(recs)
```

Comment:
@h-kataria  I think that you cant select multiple VM for Right-size Recommendations, not only that JS validation doesn't allow you to select more, but if you look at `right_size` method in `vm_common.rb` we had this piece of code there ` @record = Vm.find_by_id(params[:id])`. So the Right-size as it is now, work only for 1 VM, I a not sure if it's bug or feature, but it was like that and I am only make it more explicit.